ethereum/core/src/main/java/org/hyperledger/besu/ethereum/util/LogUtil.java

```diff
@@ -0,0 +1,26 @@
+package org.hyperledger.besu.ethereum.util;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Consumer;
+
+public class LogUtil {
+  public static void throttledLog(
+      final Consumer<String> logger,
+      final String logMessage,
+      final AtomicBoolean shouldLog,
+      final int logRepeatDelay) {
+    if (shouldLog.compareAndSet(true, false)) {
+      logger.accept(logMessage);
+      final ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
```

Comment:
If we end up calling `throttledLog` from multiple places, could we end up with performance issues with too many threads?

Reply:
I agree creating the executor on every invocation is overkill in terms of performance, you can use the `ForkJoinPool.commonPool` here